client/src/states/auth/login/controller.js

```diff
@@ -1,21 +1,16 @@
-function loginController($http, $state) {
+function loginController($http, $state, AuthService) {
     let self = this;
     self.loginErrorMessage = false;
     self.username = '';
     self.password = '';
-    self.user = '';
+    self.user = null;
 
     self.logIn = function () {
         var data = {
         username: self.username,
         password: self.password
```

Comment:
Keep in mind that HTTP is an open text protocol, so a common practice is no to pass passwords in clear form, but to use password hash and check a hash on server side. Or to use HTTPS encrypted connection.